File: 3128. Right Triangles.py
class Solution:
    def numberOfRightTriangles(self, grid: List[List[int]]) -> int:
        colOnes = []
        rowOnes = []
        for i in range(len(grid)):
            count = 0
            for j in range(len(grid[i])):
                count += grid[i][j]
            rowOnes.append(count)
        for j in range(len(grid[0])):
            count = 0
            for i in range(len(grid)):
                count += grid[i][j]
            colOnes.append(count)
        answer = 0
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                if colOnes[j] <= 1 or rowOnes[i] <= 1 or grid[i][j] == 0:
                    continue
                answer += (colOnes[j]-1) * (rowOnes[i]-1)
        return answer

File: 3128. Right Triangles.java
class Solution {
    public long numberOfRightTriangles(int[][] grid) {
        int n = grid.length;
        int m = grid[0].length;
        int[] colOnes = new int[m];
        int[] rowOnes = new int[n];
        for (int i = 0; i < n; i++){
            int count = 0;
            for (int j = 0; j < m; j++) {
                count += grid[i][j];
            }
            rowOnes[i] = count;
        }
        for (int j = 0; j < m; j++) {
            int count = 0;
            for (int i = 0; i < n; i++) {
                count += grid[i][j];
            }
            colOnes[j] = count;
        }
        long answer = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (colOnes[j] <= 1 || rowOnes[i] <= 1 || grid[i][j] == 0) continue;
                answer += (long)(colOnes[j]-1)*(rowOnes[i]-1);
            }
        }
        return answer;
    }
}